platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/DefaultProperty.java

```diff
@@ -24,6 +24,12 @@
 
 import javax.annotation.Nullable;
 
+/**
+ * The implementation for general-purpose (atomic) properties, where
```

Comment:
💅 What does "atomic" mean here?  We should document it.

Reply:
added "i.e., non-composite"